tests/TestCase/Utility/HashTest.php

```diff
@@ -1489,6 +1490,32 @@ public function testExtractUnevenKeys()
     }
 
     /**
+     * Tests that objects as values handled correctly.
+     *
+     * @return void
+     */
+    public function testExtractObjects()
+    {
+        $data = [
+            'root' => [
+                'array' => new ArrayObject([
+                    'foo' => 'bar',
+                ]),
+                'created' => new Time('2010-01-05'),
+            ],
+        ];
+
+        $result = Hash::extract($data, 'root.created');
+        $this->assertSame([$data['root']['created']], $result);
+
+        $result = Hash::extract($data, 'root.array');
+        $this->assertSame($data['root']['array'], $result);
```

Comment:
Unless the function results are put into one of the `array_*` functions. Then its not backwards compatible.